Sources/MapboxMaps/Style/Style+Localization.swift

```diff
@@ -0,0 +1,83 @@
+@_implementationOnly import MapboxCommon_Private
+
+/// Support for localization of labels
+extension Style {
+
+    /// This function creates an expression that will localize the `textField` property of a `SymbolLayer`
+    /// - Parameter locale: A `SupportedLanguage` based `Locale`
+    internal func localizeLabels(into locale: Locale) throws {
+
+        /// Get all symbol layers that are currently on the map
+        let symbolLayers = allLayerIdentifiers.filter { layer in
+            return layer.type == .symbol
+        }
```

Comment:
Not every symbol layer should be forced to have a `name`-based `text-field`, which is specific to the Mapbox Streets source. For example, consider a symbol layer backed by a vector tileset or GeoJSON source with the labelable text in `NAME` instead of `name`. Now localizing the rest of the map will blow away any labeling of this non-Streets source.